### DISCUSSION TITLE
I was confused

### DISCUSSION THREAD

**Discussion by ucdo** [rabbitmq/rabbitmq-server#9834] [Other] [answered]
### Describe the bug

I just want  to receive message which routing_key is *.test.*,but I received all.  Maybe It  don`t use like this?

### Reproduction steps

publisher:
![image](https://github.com/rabbitmq/rabbitmq-server/assets/95133030/61acf3a8-ba54-4fa3-8af7-8368b224d99a)
consumer:
![image](https://github.com/rabbitmq/rabbitmq-server/assets/95133030/c6f7995c-0be3-430d-aa08-6ee08d279d0e)

### Expected behavior

aa.test.aa   test1
aa.test.aa   test2
aa.test.aa   test3

**Comment by michaelklishin** [comment, marked as answer]
I don't think I understand what the goal is, as `.test.` is a weird routing key since it starts with a segment separator. The segments are then `["", "test", ""]`. `*` matches a single segment in AMQP 0-9-1, including when it is empty (or at least the spec does not special case an empty segment, and neither does the RabbitMQ topic exchange routing implementation IIRC).

You bind the same queue twice with two routing patterns. Regardless of which one of them matches, the same queue will end up containing both "kinds" of messages. Usually if you want to separate messages you must also use different queues, otherwise why use different routing patterns at all.

**Reply by ucdo** [reply]
You are right. Because I bind twice in publisher,so the queue received all  the message.
